webpack/assets/javascripts/react_app/components/bookmarks/form/index.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { reduxForm } from 'redux-form';
+import { connect } from 'react-redux';
+import Form from '../../common/forms/Form';
+import TextField from '../../common/forms/TextField';
+import { required, length } from 'redux-form-validators';
+import * as FormActions from '../../../redux/actions/common/forms';
+
+class BookmarkForm extends React.Component {
+  // if we got here, its already post client validations
+  submit({ name, query, publik }, dispatch, props) {
+    const { submitForm, url, controller } = props;
+    const values = { name, query, public: publik || false, controller };
+
+    return submitForm({url, values, item: 'Bookmark'});
+  }
+
+  render() {
+    const { handleSubmit, submitting, error, onCancel } = this.props;
+
+    return (
+      <Form
+        onSubmit={handleSubmit(this.submit)}
+        onCancel={onCancel}
+        disabled={submitting}
+        submitting={submitting}
+        error={error}
+      >
+        <TextField
+          name="name"
+          type="text"
+          required="true"
```

Comment:
Can we somehow reuse the existing helper that determines whether the field is requires based on model validation?

Reply:
since this is purely data driven, we could and we should probably do that, but i thought it should not be forced in the same pr.

Reply:
but take note that we don't need necessarily to map rails models -> form elements.

Reply:
if this is building a foundation for new generation of forms, I'd prefer to have it solved from the first day, otherwise we'd build technical debt to change it in all forms later